### PR TITLE
fixed optional dependencies for geoips_clavrx repo

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] NO REQUIRED ***unit tests*** (explain why not required)
* [x] NO REQUIRED ***integration tests*** (explain why not required)
* [x] Required ***documentation*** added for new/modified functionality
* [x] Required ***release notes*** added for new/modified functionality
* [x] Required ***updates to other repos*** complete

# Related Issues
fixes NRLMMD-GEOIPS/geoips#338
This PR branches off of another GeoIPS [PR](https://github.com/NRLMMD-GEOIPS/geoips/pull/438), which changes the implementation of optional dependencies scattered throughout the code. Don't merge this branch until the GeoIPS PR is merged.

# Testing Instructions
Run geoips/tests/integration_tests/full_test.sh to ensure this PR works. Works on my end.

# Summary
*From GEOIPS#338: 2023-07-19, Clean up optional dependencies*

Some GeoIPS readers include optional dependency statements that are required to read
certain file formats. While keeping these imports as optional is ok, we should clean up
the manner in which this is implemented. To do so, we've created a separate
geoips.testing.context_manager.py script which can handle optional imports scattered
throughout the GeoIPS codebase. This is essentially replacing our old manner of optional
dependencies with a new method that keeps things clean.

- modified: geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py

### CODE REVIEW

File: docs/source/releases/v1_12_0a0.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Version 1.12.0a0 (2024-02-12)
+*****************************
+
+* Refactor: Clean up optional dependencies throughout the codebase.
+
+Refactoring Updates
+-------------------
+
+Clean up optional dependencies throughout the code base
```
Comment (evrose54) [inline docs/source/releases/v1_12_0a0.rst:21]: Don't merge until this [PR](https://github.com/NRLMMD-GEOIPS/geoips/pull/438) has been merged.

**Review by jsolbrig — APPROVED** [review]

**Review by jsolbrig — APPROVED** [review]
This looks good to me. It should be merged as soon as the up-stream PR is merged. See the other comment.

File: docs/source/releases/v1_12_0a0.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ | # # # Naval Research Laboratory, Marine Meteorology Division
+ | # # #
+ | # # # This program is free software: you can redistribute it and/or modify it under
+ | # # # the terms of the NRLMMD License included with this program. This program is
+ | # # # distributed WITHOUT ANY WARRANTY; without even the implied warranty of
+ | # # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+ | # # # for more details. If you did not receive the license, for more information see:
+ | # # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+
+Version 1.12.0a0 (2024-02-12)
```
Comment (mindyls) [inline docs/source/releases/v1_12_0a0.rst:13]: Should this be 1.12.2a0 ?  I think it should match whatever is in geoips repo update_this_release_note file:
https://github.com/NRLMMD-GEOIPS/geoips/blob/main/update_this_release_note

File: docs/source/releases/v1_12_0a0.rst (hunk 1 above)
Comment (jsolbrig) [reply docs/source/releases/v1_12_0a0.rst:21]: Ready now.

**Review by mindyls — APPROVED** [review]
Conceptually looks good to me - I did not test it, so assuming this worked for @jsolbrig and you take a look at my conversation, I am good to merge.

File: docs/source/releases/v1_12_0a0.rst (hunk 2 above)
Comment (mindyls) [reply docs/source/releases/v1_12_0a0.rst]: Past Mindy says that it should be 1.12.2a0, based on the GitHub Actions she set up:
https://github.com/NRLMMD-GEOIPS/geoips_clavrx/actions/runs/8694314032/job/23842934175?pr=34

Comment (jsolbrig) [reply docs/source/releases/v1_12_0a0.rst]: I updated these.

**Pull request merged**